packages/core/src/actions/smartAccount/internal/sendUserOperation.ts

```diff
@@ -16,10 +17,18 @@ export const _sendUserOperation: <
   TChain extends Chain | undefined = Chain | undefined,
   TAccount extends SmartContractAccount | undefined =
     | SmartContractAccount
-    | undefined
+    | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 >(
-  client: BaseSmartAccountClient<TTransport, TChain, TAccount>,
-  args: { uoStruct: UserOperationStruct } & GetAccountParameter<TAccount>
+  client: BaseSmartAccountClient<
+    TTransport,
+    TChain,
+    TAccount,
+    TEntryPointVersion
```

Comment:
why's the client need to be generic on the entrypoint version?

Reply:
doesn't this result in potential drift between the definition of the entrypoint version here and the version on the account?